app/notification_email_test.go

```diff
@@ -45,9 +44,7 @@ func TestGetGroupMessageNotificationEmailSubjectFull(t *testing.T) {
 	translateFunc := utils.GetUserTranslations("en")
 	emailNotificationContentsType := model.EMAIL_NOTIFICATION_CONTENTS_FULL
 	subject := getGroupMessageNotificationEmailSubject(user, post, translateFunc, "http://localhost:8065", "sender", emailNotificationContentsType, true)
-	if !strings.HasPrefix(subject, expectedPrefix) {
-		t.Fatal("Expected subject line prefix '" + expectedPrefix + "', got " + subject)
-	}
+	require.True(t, strings.HasPrefix(subject, expectedPrefix), "Expected subject line prefix '"+expectedPrefix+"', got "+subject)
```

Comment:
For `strings.HasPrefix()` tests, I believe `require.Regexp()` would be appropriate.

Reply:
can you please give an example

Reply:
`require.Regexp(t, regexp.MustCompile("^" + expectedPrefix), subject)` should work :+1:

Reply:
thanks

Reply:
Not working subject contains [] characters

Expected subject line prefix '[http://localhost:8065] New Direct Message from @sender on', got [http://localhost:8065] New Direct Message from @sender on August 4, 2017

Reply:
The square brackets in `expectedPrefix` are being processed as regular expression metacharacters. You will need to use [regexp.QuoteMeta](https://godoc.org/regexp#QuoteMeta) to escape these characters like so:

```require.Regexp(t, regexp.MustCompile("^" + regexp.QuoteMeta(expectedPrefix)), subject)```

Sorry for the back and forth on this!

Reply:
done. working now.